metis-indexing/src/main/java/eu/europeana/indexing/fullbean/RdfToFullBeanConverter.java

```diff
@@ -78,26 +78,28 @@ public FullBeanImpl convertRdfToFullBean(RdfWrapper record) {
   }
 
   private List<Aggregation> convertAggregations(RdfWrapper record) {
-    final Supplier<List<WebResourceImpl>> webResourcesExtractor = new WebResourcesExtractor(record);
+    //The record web resources is reduced every time one of it's web resources gets referenced
+    final List<WebResourceImpl> recordWebResources = new ArrayList<>(new WebResourcesExtractor(record).get());
+    //The reference list is being extended every time a new web resource is referenced from an aggregator
+    final List<WebResourceImpl> referencedWebResources = new ArrayList<>(recordWebResources.size());
     //Convert the provider aggregations
```

Comment:
It's important that these go first, right? So that they get first claim for web resources. Should we make a note of this?

Reply:
yes sure I've added some short explanation